File: Java_20Q/Q8.java
import java.util.Scanner;
import java.io.*;

public class LLNode {
  public String payload;
  public LLNode next;

  LLNode(String payload, LLNode next) {
    this.payload = payload;
    this.next = next;
  }
}

public class Q8E {
  public static void main() throws IOException {
    Scanner in = new Scanner(new FileReader("words.txt"));
    LLNode base = null;
    LLNode temp;

    while (in.hasNextLine()) {
      String t = in.nextLine();
      temp = new LLNode(t, base);
      base = temp;
    }

    LLNode current = base;
    boolean flag = true;

    while (flag) {
      System.out.println(current.payload);
      current = current.next;
      if (current.next == null) {
        System.out.println(current.payload);
        flag = false;
      }
    }

  }
}

File: C++_20Q/Q8.cpp
#include <iostream>

class LLNode {

public:
  std::string payload;
  LLNode *next;

  LLNode(std::string payload, LLNode *next) {
    this->payload = payload;
    this->next = next;
  }
};

int main() {
  LLNode *base = nullptr;
  LLNode *temp;

  // Insert 0 node
  temp = new LLNode("Zero", base);
  base = temp;

  // Head Insert First
  temp = new LLNode("First", base);
  base = temp;

  // Head Insert Second
  temp = new LLNode("Second", base);
  base = temp;

  LLNode *current = base;
  while (current) {
    std::cout << current->payload << std::endl;
    current = current->next;
  }
}
